libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayEventLocationMapper.kt

```diff
@@ -0,0 +1,28 @@
+package com.mapbox.navigation.core.replay.history
+
+import android.location.Location
+import android.os.SystemClock
+import androidx.annotation.VisibleForTesting
+import java.util.Date
+
+private const val MILLIS_PER_SECOND = 1000.0
+private const val NANOS_PER_SECOND = 1e+9
+private fun Double.secToMillis(): Long = (this * MILLIS_PER_SECOND).toLong()
+private fun Double.secToNanos(): Long = (this * NANOS_PER_SECOND).toLong()
+
+internal fun ReplayEventLocation.mapToLocation(
+    eventTimeOffset: Double = time ?: 0.0,
+    @VisibleForTesting currentTimeMilliseconds: Long = Date().time,
+    @VisibleForTesting elapsedTimeNano: Long = SystemClock.elapsedRealtimeNanos()
+): Location {
```

Comment:
@kmadsen , I didn't understand which problems `mapboxReplayer.eventRealtimeOffset` solves, so maybe it isn't the best API for the mapper. Any suggestions?